.github/workflows/build_and_test.yaml

```diff
@@ -12,68 +12,53 @@ on:
       - "release-v*"
     paths-ignore:
       - "**/*.png"
-env:
-  GO_VERSION: 1.18.2
-  ENVOY_GATEWAY_DEV_IMAGE: envoyproxy/gateway-dev
-  ENVOY_GATEWAY_DEV_TAG: latest
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./tools/github-actions/setup-deps
+
+      # lint
+      - run: make lint-deps
+      - run: make -k lint
+
   build-and-test:
     runs-on: ubuntu-latest
     steps:
-      - name: Check out code
-        uses: actions/checkout@v3
-      - name: Run whitenoise lint
-        run: |
-          MAKE_IN_DOCKER=1 make lint.whitenoise
-      - name: Run yamllint
-        run: |
-          MAKE_IN_DOCKER=1 make lint.yamllint
-      - name: Run codespell
-        run: |
-          MAKE_IN_DOCKER=1 make lint.codespell
-      - name: Run golangci-lint 
-        run: |
-          MAKE_IN_DOCKER=1 make lint.golint
-      - name: Set up Go
-        uses: actions/setup-go@v3
-        with:
-          go-version: ${{ env.GO_VERSION }}
-      - name: Build Binaries
-        run: make build-multiarch
-      - name: Test and report coverage
-        run: make go.test.coverage
+      - uses: actions/checkout@v3
+      - uses: ./tools/github-actions/setup-deps
+
+      # build
+      - run: make build-multiarch
+
+      # test
+      - run: make go.test.coverage
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v2
         with:
           fail_ci_if_error: true
           files: ./coverage.xml
           name: codecov-envoy-gateway
           verbose: true
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
-      - name: Build Docker Image
-        uses: docker/build-push-action@v3
-        with:
-          file: ./tools/docker/envoy-gateway/Dockerfile
-          context: bin
-          platforms: linux/amd64
-          tags: ${{ env.ENVOY_GATEWAY_DEV_IMAGE }}:${{ github.sha }}
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
+
+      # push
       - name: Login to DockerHub
         if: github.event_name == 'push'
         uses: docker/login-action@v2
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_PASSWORD }}
-      - name: Push to envoyproxy/gateway-dev
+      - run: make push-multiarch
+        if: github.event_name == 'push'
+      - name: Push to :latest
         if: github.event_name == 'push' && github.ref == 'refs/heads/main'
         uses: docker/build-push-action@v3
         with:
           file: ./tools/docker/envoy-gateway/Dockerfile
           context: bin
           platforms: linux/amd64,linux/arm64
           push: true
-          tags: ${{ env.ENVOY_GATEWAY_DEV_IMAGE }}:${{ github.sha }}, ${{ env.ENVOY_GATEWAY_DEV_IMAGE }}:${{ env.ENVOY_GATEWAY_DEV_TAG }}
+          tags: docker.io/envoyproxy/gateway-dev:latest
```

Comment:
we also want to tag and push with the SHA of the commit, can you please revert this

Reply:
The sha gets pushed by the earlier `make push-multiarch`.

Reply:
ah thanks for pointing it out.
If we left this the way it is, it would be make the CI faster since its leveraging the same cache for the multiple build steps

Reply:
IMO, since everything should be doable without GHA, the better solution would be to go the other direction and move the `:latest` pushing to be a target in the Makefile, but I didn't have the appetite to do that in this PR.

Reply:
would prefer if this GHA remains the way it is, since the current changes doesn't improve the existing workflow and keep things consistent

Reply:
@kflynn that sounds like a valid case, prefer if we handles ^ Makefile target in a separate PR once completed so it does not overload this PR which is addressing https://github.com/envoyproxy/gateway/issues/124